conda_devenv/devenv.py

```diff
@@ -439,8 +456,11 @@ def _call_conda(env_manager: str = "conda"):
 
 
 def write_activate_deactivate_scripts(
-    args, conda_yaml_dict, environment, env_directory
-):
+    args: argparse.Namespace,
+    conda_yaml_dict: YAMLData,
+    environment: Environment,
+    env_directory: Path | None,
+) -> None:
     if env_directory is None:
         env_name = args.name or conda_yaml_dict["name"]
```

Comment:
This is still going to blow up if `env_directory` is `None`, `args.name` is falseable and `conda_yaml_dict` is the empty dict (as passed on line 709), I'm not sure if we should add an explicit check, or maybe improve the signature (or add overloads, but `argparse.Namespace` complicates it).

Reply:
I will raise an error here, I think it is fair to explicitly tell the user to either configure `name` in the YAML or via `--name`.